drivers/sensors/ms5611.c

```diff
@@ -0,0 +1,702 @@
+ * interval (-i) parameter, so let keep it comment until further
+ * discussion about the "issue".
+ */
+
+#if 0
+static int ms5611_fetch(FAR struct sensor_lowerhalf_s *lower,
+                        FAR char *buffer, size_t buflen)
+{
+  FAR struct ms5611_dev_s *priv = container_of(lower,
+                                               FAR struct ms5611_dev_s,
+                                               sensor_lower);
+  struct sensor_event_baro baro_data;
+
+  if (buflen != sizeof(baro_data))
+    {
+      return -EINVAL;
+    }
+
+  baro_measure_read(priv, &baro_data);
+
+  memcpy(buffer, &baro_data, sizeof(baro_data));
+
+  return buflen;
+}
+#endif
+
+/****************************************************************************
+ * Public Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: ms5611_register
+ *
+ * Description:
+ *   Register the MS5611 character device
+ *
+ * Input Parameters:
+ *   devno   - Instance number for driver
+ *   i2c     - An instance of the I2C interface to use to communicate with
+ *             MS5611
+ *
+ * Returned Value:
+ *   Zero (OK) on success; a negated errno value on failure.
+ *
+ ****************************************************************************/
+
+int ms5611_register(FAR struct i2c_master_s *i2c, int devno, uint8_t addr)
+{
+  FAR struct ms5611_dev_s *priv;
+  FAR char *argv[2];
+  char arg1[32];
+
+  int ret;
+
+  /* Initialize the MS5611 device structure */
+
+  priv = (FAR struct ms5611_dev_s *)kmm_zalloc(sizeof(struct ms5611_dev_s));
+  if (!priv)
+    {
+      snerr("Failed to allocate instance\n");
+      return -ENOMEM;
+    }
+
+  priv->i2c      = i2c;
+  priv->addr     = addr;
+  priv->freq     = CONFIG_MS5611_I2C_FREQUENCY;
+  priv->interval = 1000000; /* Default interval 1s */
+
+  nxsem_init(&priv->run, 0, 0);
+  nxsem_init(&priv->exclsem, 0, 1);
+  nxsem_set_protocol(&priv->run, SEM_PRIO_NONE);
+  nxsem_set_protocol(&priv->exclsem, SEM_PRIO_NONE);
+
+  priv->sensor_lower.ops = &g_sensor_ops;
+  priv->sensor_lower.type = SENSOR_TYPE_BAROMETER;
+
+  ret = ms5611_initialize(priv);
+  if (ret < 0)
+    {
+      snerr("Failed to initialize physical device ms5611:%d\n", ret);
+      kmm_free(priv);
+      return ret;
+    }
+
+  /* Register the character driver */
+
+  ret = sensor_register(&priv->sensor_lower, devno);
+  if (ret < 0)
+    {
+      snerr("Failed to register driver: %d\n", ret);
+      kmm_free(priv);
+    }
+
+  /* Create thread for polling sensor data */
+
+  snprintf(arg1, 16, "0x%" PRIxPTR, (uintptr_t)priv);
+  argv[0] = arg1;
+  argv[1] = NULL;
+  ret = kthread_create("ms5611_thread", SCHED_PRIORITY_DEFAULT,
+                       CONFIG_MS5611_THREAD_STACKSIZE,
+                       ms5611_thread, argv);
+  if (ret > 0)
+    {
+      return OK;
+    }
+
+  sninfo("MS5611 driver loaded successfully!\n");
```

Comment:
I do not fully understand this. Should `ms5611_register` always return `OK`?

Reply:
Good catch, it is in fact wrong! Thanks!